playbooks/resize_disk.yaml

```diff
@@ -0,0 +1,41 @@
+---
+- hosts: all
+  become: true
+  tasks:
+    - name: Find root partition
+      shell: df --output=source,fstype / | tail -n 1
+      register: root_partition
+
+    - name: Install cloud-utils-growpart package
+      package:
+        name: cloud-utils-growpart
+      when: ansible_os_family == 'RedHat'
+
+    - name: Install cloud-guest-utils package
+      package:
+        name: cloud-guest-utils
+      when: ansible_os_family == 'Debian'
+
+    - name: Install e2fsprogs package
+      package:
+        name: e2fsprogs
+      when: '"ext4" in root_partition.stdout'
+
+    - name: Extend partition
+      command: growpart /dev/vda {{ root_partition.stdout.split(' ')[0].replace('/dev/vda', '') }}
+      register: output
+      ignore_errors: true
+
+    - name: Fail if growpart fails, do not fail if NOCHANGE
+      fail:
+      when:
+        - '"NOCHANGE" not in output.stdout'
+        - output.rc != 0
```

Comment:
Instead of using `fail` (without any message) here, wouldn't it be possible to make a more correct [`failed_when` in the `command` step](https://docs.ansible.com/ansible/latest/user_guide/playbooks_error_handling.html#defining-failure) above?

something like:

```yaml
    - name: Extend partition
      command: growpart /dev/vda {{ root_partition.stdout.split(' ')[0].replace('/dev/vda', '') }}
      register: growpart_output
      failed_when:
        - '"NOCHANGE" not in growpart_output.stdout'
        - growpart_output.rc != 0
```

BTW, are the extra quotes around the `… not in …` required?

Reply:
Yes, the quotes appear to be needed.

Reply:
:crying_cat_face: